client/js/restaurantsAPI.js

```diff
@@ -0,0 +1,1279 @@
+'use strict';
```

Comment:
this source code file is too big

Why don't you divide it by functionality, for instance: 

restaurant list
restaurant data edition
reservations 

... 

or other criteria?